src/__tests__/404/404.test.ts

```diff
@@ -0,0 +1,62 @@
+import { getPage } from '../../../src';
+import path from 'path';
+import { screen } from '@testing-library/react';
+import { stripReactExtraMarkup } from '../__utils__/expectDOMElementsToMatch';
+
+describe('404', () => {
+  describe.each([
+    ['ssr "notFound" return', '/a'],
+    // ['matching page not found', '/random'],
```

Comment:
Tests stayed the same, but were wrapped in top level `404` describe so they can be reused. When implemented, the lined with `'matching page not found'` can be un-commented and tests should pass